src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -167,26 +151,61 @@ public virtual async Task<HttpWebResponseWrapper> SendAsync()
                         {
-                            if (response != null && response.StatusCode == HttpStatusCode.PartialContent)
+                            // HttpClient.SendAsync throws HttpRequestException only on the following scenarios:
+                            // "The request failed due to an underlying issue such as network connectivity, DNS failure, server certificate validation or timeout."
+                            // i.e for Server errors (500 status code), no exception is thrown. Hence this method should read the response and status code,
+                            // and return correct HttpWebResponseWrapper to give any Retry policies a chance to retry as needed.
+
+                            using (var response = await client.SendAsync(request, ct.Token).ConfigureAwait(false))
                             {
-                                wrapper = new HttpWebResponseWrapper
-                                {
-                                    StatusCode = (int)response.StatusCode,
-                                    StatusDescription = response.ReasonPhrase
-                                };
-                                wrapper.RetryAfterHeader = response.Headers?.RetryAfter?.ToString();
-                                if (response.Content != null)
+                                if (response != null)
                                 {
-                                    wrapper.Content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                                    wrapper = new HttpWebResponseWrapper
+                                    {
+                                        StatusCode = (int)response.StatusCode,
+                                        StatusDescription = response.ReasonPhrase // maybe not required?
+                                    };
+                                    wrapper.RetryAfterHeader = response.Headers?.RetryAfter?.ToString();
+
+                                    if (response.StatusCode == HttpStatusCode.PartialContent)
+                                    {
+                                        if (response.Content != null)
+                                        {
+                                            wrapper.Content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                                        }
+                                    }
+
+                                    if (CoreEventSource.IsVerboseEnabled)
+                                    {
+                                        // Read the entire response body only on VerboseTracing for perf reasons.
+                                        try
+                                        {
+                                            if (response.Content != null)
+                                            {
+                                                wrapper.Content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```

Comment:
If it was `PartialContent` status code before on line 174, then `wrapper.Content` may already contain response boxy as per 178, so we may skip reading it the second time in this case.